src/webview_candidate_window.mm

```diff
@@ -36,6 +36,74 @@ - (BOOL)isKeyWindow {
 
 @end
 
+#ifdef WKWEBVIEW_PROTOCOL
+@implementation FileSchemeHandler
+
+// In f5m, it serves fcitx:///file/foo/bar from
+// ~/.local/share/fcitx5/www/foo/bar
+- (void)webView:(WKWebView *)webView
+    startURLSchemeTask:(id<WKURLSchemeTask>)urlSchemeTask {
+    NSURL *url = urlSchemeTask.request.URL;
+    if (!url || ![url.path hasPrefix:@"/file"]) {
+        [urlSchemeTask didFailWithError:[NSError errorWithDomain:@"Invalid URL"
```

Comment:
I think "F5mErrorDomain" is better.

Reply:
Sure, it also inform others which IME first uses it 🌚